code/modules/mob/living/carbon/human/human_defense.dm

```diff
@@ -162,6 +162,13 @@
 	else
 		affecting = get_bodypart(ran_zone(user.zone_selected))
 	var/target_area = parse_zone(check_zone(user.zone_selected)) //our intended target
+	if(I.force && I.damtype != STAMINA && affecting.status == BODYPART_ROBOTIC) // Bodpart_robotic sparks when hit, but only when it does real damage
+		var/datum/effect_system/spark_spread/spark_system = new /datum/effect_system/spark_spread
+		spark_system.set_up(1, 0, src)
+		spark_system.attach(src)
+		spark_system.start()

```

Comment:
whoaaaaa gross - no no no.

You're building and setting up a new thing each time!!!!

Create it once during `Initialize` then start it when you need to trigger it.

*scream *scream *scream